File: Basic Exercises/Normal/2/2. Add Two Numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *a=l1,*b=l2;
        int va,vb,newv,t=0;
        auto h= new ListNode(0);
        auto na=h;
        while(a!=nullptr or b!=nullptr){
            if(a!=nullptr){
                va=a->val;
                a=a->next;
            }else{
                va=0;
            }
            if(b!=nullptr){
                vb=b->val;
                b=b->next;
            }else{
                vb=0;
            }
            newv=va+vb+t;
            if(newv<10){
                t=0;
            }else{
                t=1;
                newv=newv-10;
            }
            auto node=new ListNode(newv);
            na->next=node;
            na=node;
        }
        if(t==1){
            auto node=new ListNode(1);
            na->next=node;
        }
        return h->next;
    }
};

File: Basic Exercises/Normal/2/2. Add Two Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        a=l1
        b=l2
        t=0
        h=ListNode(0)
        na=h
        while(a or b):
            va,a=(a.val,a.next) if a else (0,a)
            vb,b=(b.val,b.next) if b else (0,b)
            new=va+vb+t
            new,t=(new,0) if new<10 else (new-10,1)
            node=ListNode(new)
            na.next,na=node,node
        if t==1:
            node=ListNode(1)
            na.next=node
        return h.next
